### DISCUSSION TITLE
export or conversion:  slides <--> jupyter notebook <--> whiteboard

### DISCUSSION THREAD

**Discussion by williamstein** [sagemathinc/cocalc#6427] [Ideas]
CoCalc has [slides](https://cocalc.com/features/slides), [Jupyter notebooks](https://cocalc.com/features/jupyter-notebook), and [whiteboards](https://cocalc.com/features/whiteboard).  

It would be natural implement conversions between these formats, at least in various cases.  This discussion is just about mainly gathering thoughts or ideas about what that might look like, or what people are interested in.

**Comment by mforbes** [comment]
There are a couple of reasons that I currently think I prefer using Jupyter Notebooks for slides:

1. It is important for me to be able to edit/show my slides offline (i.e. when working on an airplane, or to be sure that network connectivity will not be an issue during a talk).  I don't see obviously how to do this with slides.
2. I kind of like not having to think about layout initially: i.e. the lack of flexibility when using Jupyter Notebooks.  That being said, being able to arrange things manually now and then could certainly be nice.  Is there some way of defining "formats" with slides that allows one to ignore or reset customizations so that thinking about form and content can be separated?

**Comment by williamstein** [comment]
> It is important for me to be able to edit/show my slides offline [...] I don't see obviously how to do this with slides.

Use cocalc-docker:  https://github.com/sagemathinc/cocalc-docker#cocalc-docker-image

The latest image updated today has presentation mode, etc.

> I kind of like not having to think about layout initially

This absolutely makes good sense, and I really like the idea of a workflow that starts with just the structure, and ends with fine tuning the visual look...
